### PR TITLE
fx quant: fix types on _find_quants

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49717 fx quant: hook up ConvTranspose{n}d
* #49688 fx quant: improve types on convert
* #49628 fx quant: do not observe bias on F.linear
* #49623 fx quant: do not observe bias on F.conv
* **#49616 fx quant: fix types on _find_quants**
* #49607 fx quant: add types to observed_module.py
* #49606 fx quant: types for fusion_patterns.py
* #49549 unbreak mypy torch/quantization

Summary:

Add types to `_find_quants` I/O and fix resulting errors,
needed for an upcoming bug fix.

Test Plan:

```
mypy torch/quantization
python test/test_quantization.py TestQuantizeFx
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25645719](https://our.internmc.facebook.com/intern/diff/D25645719)

### CODE REVIEW

File: torch/quantization/fx/quantize.py [hunk 1]
```diff
@@ -981,7 +985,7 @@ def visit_arg(arg):
                     # overwrite previous activation post process constructor if
                     # necessary
                     quants[arg.name] = (
-                        DefaultQuantizeHandler(self, arg), act_post_process_ctr)
+                        DefaultQuantizeHandler(self, arg), act_post_process_ctr, is_weight)
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:988]: I think we don't need is_weight, this info is used to decide act_post_process_ctr already in L976

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py:988]: also we can remove L978-979 I think

File: torch/quantization/fx/quantize.py [hunk 2]
```diff
@@ -956,7 +959,8 @@ def _find_quants(self, graph: Graph, matches: Dict[str, MatchResult],
          node_name -> (QuantizeHandler instance (always DefaultQuantizeHandler),
          activation_post_process (observer/fake_quantize module) constructor)
         """
-        quants: Dict[str, Any] = {}
+        # quants: Dict[str, Any] = {}
+        quants: Dict[str, Tuple[DefaultQuantizeHandler, Callable, bool]] = {}
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:963]: we can remove `bool` here

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
as discussed in chat, we can remove `DefaultQuantizeHandler`, thanks!

File: torch/quantization/fx/quantize.py [hunk 3]
```diff
@@ -571,14 +574,9 @@ def load_non_quantized(n):
             return env[n.name]
 
         def load_quantized(n):
-            if n.name not in quant_env:
```
Comment (vkuzo) [inline torch/quantization/fx/quantize.py:574]: this code was flagged unreachable, deleting

**Comment by vkuzo** [comment]
> as discussed in chat, we can remove DefaultQuantizeHandler, thanks!

separate PR, to keep things small? since this is really just annotating types and fixing bugs uncovered by that

**Review by jerryzh168 — APPROVED** [review]

**Pull request closed without merging**